audformat/core/database.py

```diff
@@ -558,6 +607,22 @@ def join_field(
                 else:
                     self[table_id] = table.copy()
 
+        # copy media files
+
+        if copy_media and self.root is not None:
+            for other in others:
+                if len(other.files) > 0 and other.root is None:
+                    raise RuntimeError(
+                        f"You can only use update with saved databases. "
+                        f"The database '{other.name}' was not saved yet."
+                    )
+                for file in other.files:
+                    src_file = os.path.join(other.root, file)
+                    dst_file = os.path.join(self.root, file)
+                    dst_dir = os.path.dirname(dst_file)
+                    audeer.mkdir(dst_dir)
+                    shutil.copy(src_file, dst_file)
```

Comment:
Before we copy, should we check if `src_file` exists and raise a custom error message in that case?

Reply:
I think the error message `shutil.copy()` returns should be enough:

```
>>> shutil.copy('non-existing.txt', '.')
FileNotFoundError                         Traceback (most recent call last)
<ipython-input-6-f7a24d9d4024> in <module>
----> 1 shutil.copy('non-existing.txt', '.')

~/.envs/audb/lib/python3.6/shutil.py in copy(src, dst, follow_symlinks)
    243     if os.path.isdir(dst):
    244         dst = os.path.join(dst, os.path.basename(src))
--> 245     copyfile(src, dst, follow_symlinks=follow_symlinks)
    246     copymode(src, dst, follow_symlinks=follow_symlinks)
    247     return dst

~/.envs/audb/lib/python3.6/shutil.py in copyfile(src, dst, follow_symlinks)
    118         os.symlink(os.readlink(src), dst)
    119     else:
--> 120         with open(src, 'rb') as fsrc:
    121             with open(dst, 'wb') as fdst:
    122                 copyfileobj(fsrc, fdst)

FileNotFoundError: [Errno 2] No such file or directory: 'non-existing.txt'
```